ports/rp2/mphalport.c

```diff
@@ -199,17 +198,32 @@ mp_uint_t mp_hal_stdout_tx_strn(const char *str, mp_uint_t len) {
     return did_write ? ret : 0;
 }
 
+void mp_hal_delay_us(mp_uint_t us) {
+    // Until soft timer supports microseconds, avoid calling sleep_us() and
+    // invoking the alarm pool by splitting long sleeps into an optional longer
+    // sleep and a shorter busy-wait
+    uint64_t end = time_us_64() + us;
+    if (us > 1000) {
+        mp_hal_delay_ms(us / 1000);
+    }
+    while (time_us_64() < end) {
+        mp_event_handle_nowait();
+    }
```

Comment:
This function was originally a busy wait, in the sense that no MP events were called.  That's to make sure that it's accurate, because handling MP events may take a long time (eg milliseconds).

stm32 simply does a busy wait for delay_us.  esp32 has a more complex implementation, but busy-waits if the delay is short.

Maybe do something similar to esp32, and only call `mp_event_handle_nowait()` if there's enough time left?

Reply:
> I think I'll just keep it simple but go the other way and busy-wait in a tight loop for up to a millisecond, if that's OK with you.

Yep, that's fine.